### DISCUSSION TITLE
How to pass row data (or the model) to the Details view?

### DISCUSSION THREAD

**Discussion by eswachsman** [Power-Components/livewire-powergrid#609] [Q&A] [answered]
I am trying to use the Details on my rows, but I cannot find information in the documentation how to pass the row (or model) information to the details view.

I am able to pass the options array, but how to I pass along row data to the view?

**Comment by dansysanalyst** [comment, marked as answer]
HI @eswachsman,

Thank you for your question. You can access your model data with the `$row` variable. 

```php
<?php

use PowerComponents\LivewirePowerGrid\Detail;

public function setUp(): array
    {
        $this->showCheckBox();

        return [
          //...

            Detail::make()
                ->view('components.detail')
                ->options(['message' => 'hello world'])
                ->showCollapseIcon(),
        ];
    }
```

And I've created a view with:

```html
<!-- File: resources/views/components/detail.blade.php -->

<div>
    <p>
        Dish has {{ $row->calories }} ({{ $options['message'] }})
    </p>
</div>
```

Resulting in:

<img width="1352" alt="CleanShot 2022-09-05 at 00 52 20@2x" src="https://user-images.githubusercontent.com/79267265/188336517-c310897a-67bf-43b0-af64-5c189b7166ca.png">

---

This example was added to [PowerGrid-Demo](https://github.com/Power-Components/powergrid-demo) repository and opened a PR to improve the documentation as it is not clear at the moment.

**Reply by eswachsman** [reply]
Great! Thanks!
